### PR TITLE
Dispatch a `sidebarviewchanged` event in `PDFSidebar` when the view changes

### PR DESCRIPTION
**Pull request by Snuffleupagus**
We cannot piggy-back on the `updateviewarea` event in order to update the stored sidebar state, since there're a number of cases where opening/switching the sidebar view won't fire a `updateviewarea` event.
Note that `updateviewarea` only fires when the position changes in the _viewer_, which means that it won't fire if e.g. the viewer is narrow, such that the sidebar overlays the document transparently; or when switching views, without the document position also changing.

This patch also moves the handling of `forceOpen` parameter in `PDFSidebar_switchView`, to prevent triggering back-to-back rendering and dispatching of events.

/cc @timvandermeij This ought to take care of the issues you pointed out in #7010.
I should probably just have added the event in #7038, but since there wasn't any consumer of it back then, it simply slipped my mind.

### CODE REVIEW

File: web/app.js [hunk 1]
```diff
@@ -1630,23 +1633,40 @@ window.addEventListener('presentationmodechanged', function (e) {
...
+  }
+  PDFViewerApplication.pdfRenderingQueue.isThumbnailViewEnabled =
+    PDFViewerApplication.pdfSidebar.isThumbnailViewVisible;
+
+  var store = PDFViewerApplication.store;
+  if (!store || !PDFViewerApplication.isInitialViewSet) {
+    // Only update the storage when the document has been loaded *and* rendered.
+    return;
+  }
+  // store.initializedPromise.then(function() {
```
Comment (timvandermeij) [inline web/app.js]: Why is this code commented out?

**Comment by Snuffleupagus** [comment]
There are parts of this PR, e.g. the writing to storage, that probably shouldn't land unless we *also* add support for getting/utilizing the `sidebarView` at the same time. Hence why I left some code commented out, since I didn't want to add unused code.

@timvandermeij One option could be that I append the commit from PR #7010 here, so that it can land all at once? I'm not sure if that makes testing more difficult though, so what do you think?

File: web/pdf_sidebar.js [hunk 2]
```diff
@@ -90,9 +90,6 @@ var PDFSidebar = (function PDFSidebarClosure() {
     reset: function PDFSidebar_reset() {
       this.isInitialViewSet = false;
 
-      this.close();
```
Comment (timvandermeij) [inline web/pdf_sidebar.js]: Why were these lines removed?

**Comment by timvandermeij** [comment]
I think appending the commit (with the right author information) here should be fine. I think testing actually becomes easier because it allows me to test both patches at once.

File: web/pdf_sidebar.js (hunk 2 above)
Comment (Snuffleupagus) [inline web/pdf_sidebar.js]: Sorry, this is probably not *entirely* relevant to this PR, and it concerns an edge-case anyway.

When looking through the code in `PDFSidebar`, it struck me that forcibly closing the sidebar during `PDFViewerApplication.close()` may not be what the user expects. This really only matters in default viewer, when the "Open" functionality is present.

If you feel that we shouldn't change this at all, or that it doesn't belong in this PR (which I'd totally understand!), I'll re-add these two lines!

**Comment by timvandermeij** [comment]
Thank you!
